src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -556,7 +556,7 @@ protected function getAbsoluteUri($uri)
         } else {
             $currentUri = sprintf('http%s://%s/',
                 isset($this->server['HTTPS']) ? 's' : '',
-                isset($this->server['HTTP_HOST']) ? $this->server['HTTP_HOST'] : 'localhost'
+                isset($this->server['HTTP_HOST']) && !empty($this->server['HTTP_HOST']) ? $this->server['HTTP_HOST'] : 'localhost'
```

Comment:
you can remove the ``isset`` check then, as ``!empty`` already takes care of the check